### PR TITLE
[cxxmodules] Don't declare strings from rootmap for modules

### PR DESCRIPTION
**Pull request by vgvassilev**
We don't need any input_line declration from rootmap for startup time. However it may cause a failure on tests when they have custom dictionaries.
    
Should give 5MB improvements for modules.
    
Patch by Yuka Takahashi and me!

### CODE REVIEW

**Comment by vgvassilev** [comment]
@phsft-bot build!

File: core/metacling/src/TCling.cxx [hunk 1]
```diff
@@ -5198,7 +5208,11 @@ namespace {
 
 Int_t TCling::LoadLibraryMap(const char* rootmapfile)
 {
+   if (rootmapfile && *rootmapfile && !requiresRootMap(rootmapfile, fInterpreter))
```
Comment (pcanal) [inline core/metacling/src/TCling.cxx]: maybe
```
rootmapfile == nullptr || *rootmapfile = '\0' || !requiresRootMap(rootmapfile, fInterpreter))
```

**Comment by vgvassilev** [comment]
@phsft-bot build just on ROOT-ubuntu16/rtcxxmod with flags -DCTEST_TEST_EXCLUDE_NONE=On

File: core/metacling/src/TCling.cxx (hunk 1 above)
Comment (vgvassilev) [reply core/metacling/src/TCling.cxx]: `requiresRootMap` assumes we have checked for `rootmapfile && *rootmapfile`

Comment (pcanal) [reply core/metacling/src/TCling.cxx]: I know :) and the proposed new version does that and return early if the rootmapfile string is empty or nullptr.

Comment (vgvassilev) [reply core/metacling/src/TCling.cxx]: Now I remember, when `rootmapfile == nullptr` we do a scan over all rootmap files. We cannot use the generic condition to exit early which I prefer, too :)

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by vgvassilev** [comment]
The failures seem unrelated.

**Comment by Axel-Naumann** [comment]
@vgvassilev how so?

**Comment by Axel-Naumann** [comment]
@phsft-bot build just on mac1014/cxx17

**Comment by Axel-Naumann** [comment]
Yup, @vgvassilev - build failures were unrelated! :-) And fixed!

**Comment by amadio** [comment]
What about Travis CI failure?

**Comment by vgvassilev** [comment]
@amadio, it also seems unrelated.

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Pull request merged**